R/linearregutils.R

```diff
@@ -82,15 +82,29 @@ df2NumericMatrix <- function(x){
 processElasticNet <- function(inputs, config){
   var_names <- getNamesFromOrdered(names(inputs$the.data), config$`Use Weight`)
   glmFun <- if (config$cv_glmnet) glmnet::cv.glmnet else glmnet::glmnet
-  x <- df2numericMatrix(inputs$the.data[,var_names$x])
+  x <- df2NumericMatrix(inputs$the.data[,var_names$x])
   funParams <- list(x = x,
                     y = inputs$the.data[,var_names$y], family = 'gaussian',
                     intercept  = config$`Omit Constant`, standardize = config$standardize_pred,
                     weights = if (!is.null(var_names$w)) inputs$the.data[,var_names$w] else NULL,
                     nfolds = if (config$cv_glmnet) config$nfolds else NULL
   )
   the.model <- do.call(glmFun, Filter(Negate(is.null), funParams))
-  the.model$lambda_pred <- config$lambda
+  if (config$cv_glmnet) {
+    #The predict function used with objects of class cv.glmnet can be
+    #called with s = "lambda.1se" or s = "lambda.min" .
+    if (config$lambda_1se) {
+      the.model$lambda_pred <- "lambda.1se"
+    } else {
+      the.model$lambda_pred <- "lambda.min"
+    }
```

Comment:
Also, this is exactly the type of logic that would be best served by writing tests. This would ensure that you don't break anything accidentally when making changes at a later stage.